tests/grids/test_mortar_grid.py

```diff
@@ -0,0 +1,1094 @@
+    # place, the lowermost are duplicated towards the end of the face
+    # definition.
+    assert np.all(new_projection[0, fi[:2]].toarray() == 0.5)
+    assert np.all(new_projection[1, fi[2:]].toarray() == 0.5)
+
+
+def test_2d_domain_replace_2d_grid_with_coarse_grid():
+    """Replace the 2d grid with a coarser one."""
+
+    # Create a first md grid
+    mdg, sd_old, old_projection = _create_2d_mdg([2, 2])
+
+    # Create a new grid. We will take the higher dimensional grid from this and insert
+    # it into the old md grid.
+    _, sd_new, _ = _create_2d_mdg([1, 2])
+
+    # Replace the 2d grid with a finer one and get the new projections. This function
+    # will also verify that the projections have the right size.
+    new_projection_avg, new_projection_int = _replace_2d_grid_fetch_projections(
+        mdg, sd_old, sd_new, old_projection
+    )
+
+    # Projection of averages sum to unity in the rows.
+    assert np.all(new_projection_avg.toarray().sum(axis=1) == 1)
+    # Columns in integrated projection sum to either 0 or 1
+    assert np.all(
+        np.logical_or(
+            new_projection_int.A.sum(axis=0) == 1,
+            new_projection_int.A.sum(axis=0) == 0,
+        ),
+    )
+
+    fi = np.where(sd_new.face_centers[1] == 0.5)[0]
+    assert fi.size == 2
+    # Hard coded test (based on knowledge of how the grids and pp.meshing
+    # is implemented). Faces to the uppermost cell are always kept in
+    # place, the lowermost are duplicated towards the end of the face
+    # definition.
+    assert np.all(new_projection_avg[0, fi[0]] == 1)
+    assert np.all(new_projection_avg[1, fi[0]] == 1)
+    assert np.all(new_projection_avg[2, fi[1]] == 1)
+    assert np.all(new_projection_avg[3, fi[1]] == 1)
+
+
+def test_2d_domain_replace_2d_grid_with_fine_perturbed_grid():
+    """Replace the 2d grid with a finer one, and move the nodes along the interface so
+    that areas along the interface are no longer equal.
+    """
+
+    # Create a first md grid
+    mdg, sd_old, old_projection = _create_2d_mdg([1, 2])
+
+    # Create a new grid. We will take the higher dimensional grid from this and insert
+    # it into the old md grid.
+    _, sd_new, _ = _create_2d_mdg([2, 2])
+
+    # By construction of the split grid, we know that the nodes at
+    # (0.5, 0.5) are no 5 and 6, and that no 5 is associated with the
+    # face belonging to the lower cells.
+    # Move node belonging to the lower face
+    sd_new.nodes[0, [5, 6]] = [0.2, 0.7]
+    sd_new.compute_geometry()
+
+    # Replace the 2d grid with a finer one and get the new projections. This function
+    # will also verify that the projections have the right size.
+    new_projection_avg, new_projection_int = _replace_2d_grid_fetch_projections(
+        mdg, sd_old, sd_new, old_projection
+    )
+
+    # Projection of averages sum to unity in the rows
+    assert np.all(new_projection_avg.toarray().sum(axis=1) == 1)
```

Comment:
The checks on unity sums and sums to 0 or 1 could be outsourced to helper function (not specific to the each test)

Reply:
Thanks.